language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/EnumClass.scala

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.codegen.backend.java.inner
+
+import com.daml.ledger.javaapi
+import com.digitalasset.daml.lf.data.Ref.Identifier
+import com.digitalasset.daml.lf.iface
+import com.squareup.javapoet._
+import com.typesafe.scalalogging.StrictLogging
+import javax.lang.model.element.Modifier
+
+private[inner] object EnumClass extends StrictLogging {
+
+  def generate(
+      className: ClassName,
+      identifier: Identifier,
+      enum: iface.Enum,
+  ): TypeSpec = {
+
+    logger.info("Start")
+    val enumType = TypeSpec.enumBuilder(className).addModifiers(Modifier.PUBLIC)
+    enum.constructors.foreach(enumType.addEnumConstant)
```

Comment:
Usually enum constants in Java are all uppercase by convention, with words separated by `_`.

Reply:
Note that we're free to uppercase. However I wouldn't start automatically adding `_`s...

Reply:
I uppercase without `_`. -- it was more painfull that expected ;)

I found that `_`-separated format nicer but I found two uses:
- The code is quite tricky (maybe I could use Guava's [CaseFormat](https://guava.dev/releases/20.0/api/docs/com/google/common/base/CaseFormat.html)
- We can start to have conflict if user starts to use `_` it is daml-lf ids.

Reply:
Yes, let's not do underscore mangling in light of the conflict argument.